src/dc_msg.rs

```diff
@@ -709,81 +718,86 @@ pub unsafe fn dc_msg_get_text(msg: *const dc_msg_t) -> *mut libc::c_char {
 }
 
 pub unsafe fn dc_msg_get_filename(msg: *const dc_msg_t) -> *mut libc::c_char {
-    let mut ret: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut pathNfilename: *mut libc::c_char = 0 as *mut libc::c_char;
+    let mut ret = 0 as *mut libc::c_char;
+
     if !(msg.is_null() || (*msg).magic != 0x11561156i32 as libc::c_uint) {
-        pathNfilename = dc_param_get((*msg).param, DC_PARAM_FILE as i32, 0 as *const libc::c_char);
-        if !pathNfilename.is_null() {
-            ret = dc_get_filename(pathNfilename)
+        if let Some(file) = (*msg).param.get(Param::File) {
+            let file_c = to_cstring(file);
+            ret = dc_get_filename(file_c);
+            free(file_c as *mut _);
         }
     }
-    free(pathNfilename as *mut libc::c_void);
-    return if !ret.is_null() {
+    if !ret.is_null() {
         ret
     } else {
         dc_strdup(0 as *const libc::c_char)
-    };
+    }
 }
 
 pub unsafe fn dc_msg_get_filebytes(msg: *const dc_msg_t) -> uint64_t {
-    let mut ret: uint64_t = 0i32 as uint64_t;
-    let mut file: *mut libc::c_char = 0 as *mut libc::c_char;
+    let mut ret = 0;
+
     if !(msg.is_null() || (*msg).magic != 0x11561156i32 as libc::c_uint) {
-        file = dc_param_get((*msg).param, DC_PARAM_FILE as i32, 0 as *const libc::c_char);
-        if !file.is_null() {
-            ret = dc_get_filebytes((*msg).context, file)
+        if let Some(file) = (*msg).param.get(Param::File) {
+            let file_c = to_cstring(file);
+            ret = dc_get_filebytes((*msg).context, file_c);
+            free(file_c as *mut _);
         }
     }
-    free(file as *mut libc::c_void);
 
     ret
 }
 
 pub unsafe fn dc_msg_get_width(msg: *const dc_msg_t) -> libc::c_int {
     if msg.is_null() || (*msg).magic != 0x11561156i32 as libc::c_uint {
-        return 0i32;
+        return 0;
     }
 
-    dc_param_get_int((*msg).param, DC_PARAM_WIDTH as i32, 0)
+    (*msg).param.get_int(Param::Width).unwrap_or_default()
 }
 
 pub unsafe fn dc_msg_get_height(msg: *const dc_msg_t) -> libc::c_int {
-    if msg.is_null() || (*msg).magic != 0x11561156i32 as libc::c_uint {
-        return 0i32;
+    if msg.is_null() || (*msg).magic != 0x11561156 as libc::c_uint {
```

Comment:
why is this magic number being changed?  and if changed at all why not to a constant?